extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/KafkaTuningConfig.java

```diff
@@ -288,76 +76,74 @@ public KafkaTuningConfig withBasePersistDirectory(File dir)
   }
 
   @Override
-  public boolean equals(Object o)
+  public KafkaTuningConfig copyOf()
   {
-    if (this == o) {
-      return true;
-    }
-    if (o == null || getClass() != o.getClass()) {
-      return false;
-    }
-    KafkaTuningConfig that = (KafkaTuningConfig) o;
-    return maxRowsInMemory == that.maxRowsInMemory &&
-           maxRowsPerSegment == that.maxRowsPerSegment &&
-           maxBytesInMemory == that.maxBytesInMemory &&
-           Objects.equals(maxTotalRows, that.maxTotalRows) &&
-           maxPendingPersists == that.maxPendingPersists &&
-           reportParseExceptions == that.reportParseExceptions &&
-           handoffConditionTimeout == that.handoffConditionTimeout &&
-           resetOffsetAutomatically == that.resetOffsetAutomatically &&
-           Objects.equals(intermediatePersistPeriod, that.intermediatePersistPeriod) &&
-           Objects.equals(basePersistDirectory, that.basePersistDirectory) &&
-           Objects.equals(indexSpec, that.indexSpec) &&
-           Objects.equals(segmentWriteOutMediumFactory, that.segmentWriteOutMediumFactory) &&
-           Objects.equals(intermediateHandoffPeriod, that.intermediateHandoffPeriod) &&
-           logParseExceptions == that.logParseExceptions &&
-           maxParseExceptions == that.maxParseExceptions &&
-           maxSavedParseExceptions == that.maxSavedParseExceptions;
+    return new KafkaTuningConfig(
+        getMaxRowsInMemory(),
+        getMaxBytesInMemory(),
+        getMaxRowsPerSegment(),
+        getMaxTotalRows(),
+        getIntermediatePersistPeriod(),
+        getBasePersistDirectory(),
+        0,
```

Comment:
I see this class assumes `maxPendingPersists` is always 0. Would you tell me why?

Reply:
`maxPendingPersist` is being deprecated, I think it's fine to remove it completely

Reply:
`maxPendingPersists` is being deprecated, in the constructor of the original `KafkaTuningConfig` it is always set to 0

Reply:
I think setting to 0 is a bug even though it's deprecated. Please set the proper input value. The default value should be 0 if it's not provided.

Reply:
changed